packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx

```diff
@@ -90,11 +90,13 @@ export class MaskedTextField extends BaseComponent<ITextFieldProps, IMaskedTextF
   }
 
   public componentWillReceiveProps(newProps: ITextFieldProps) {
-    if (newProps.mask !== this.props.mask) {
+    if (newProps.mask !== this.props.mask || newProps.value !== this.props.value) {
       this._maskCharData = parseMask(newProps.mask, newProps.maskFormat);
-      this.state = {
+      newProps.value && this.setValue(newProps.value);
```

Comment:
if newProps.value went from "sdfasfdsf" to "", this wouldn't execute.

Reply:
this should be comparing `newProps.value !== undefined`

Reply:
Be sure to update the constructor with the same logic/check:
https://github.com/OfficeDev/office-ui-fabric-react/blob/be33ed02d7e712e59c43d56e7580c288aa488ffe/packages/office-ui-fabric-react/src/components/TextField/MaskedTextField/MaskedTextField.tsx#L82